File: src/main/java/net/redstonecraft/lanservers/LanServersPlugin.java
package net.redstonecraft.lanservers;

import java.io.IOException;
import java.net.*;
import java.nio.charset.StandardCharsets;
import java.util.Timer;
import java.util.TimerTask;

public interface LanServersPlugin {

    int getPort();
    String getMotd();

    default void enable() {
        try {
            Internal.socket = new DatagramSocket();
        } catch (SocketException ignored) {}
        Internal.timer.schedule(new TimerTask() {
            @Override
            public void run() {
                byte[] data = ("[MOTD]" +  getMotd() + "[/MOTD][AD]" + getPort() + "[/AD]").getBytes(StandardCharsets.UTF_8);
                DatagramPacket packet = new DatagramPacket(data, data.length, Internal.multicastAddress, Internal.multicastPort);
                try {
                    Internal.socket.send(packet);
                } catch (IOException ignored) {}
            }
        }, 0, 1500);
    }

    default void disable() {
        Internal.timer.cancel();
        Internal.socket.close();
    }

    class Internal {
        static Timer timer = new Timer();
        static InetAddress multicastAddress;
        static int multicastPort = 4445;
        static DatagramSocket socket;

        static {
            try {
                multicastAddress = InetAddress.getByName("224.0.2.60");
            } catch (UnknownHostException ignored) {}
        }
    }

}

File: src/main/kotlin/net/redstonecraft/lanservers/LanServersPlugin.kt
package net.redstonecraft.lanservers

import java.net.DatagramPacket
import java.net.DatagramSocket
import java.net.InetAddress
import java.util.*

interface LanServersPlugin {

    val port: Int
    val motd: String

    companion object {
        private val timer = Timer()
        private val multicastAddress = InetAddress.getByName("224.0.2.60")
        private val multicastPort = 4445
        private lateinit var socket: DatagramSocket
    }

    fun enable() {
        socket = DatagramSocket()
        timer.schedule(object: TimerTask() {
            override fun run() {
                val data = "[MOTD]$motd[/MOTD][AD]$port[/AD]".encodeToByteArray()
                val packet = DatagramPacket(data, data.size, multicastAddress, multicastPort)
                socket.send(packet)
            }
        }, 0, 1500)
    }

    fun disable() {
        timer.cancel()
        socket.close()
    }

}
